apps/web/src/pages/templates/hooks/usePreviewSmsTemplate.ts

```diff
@@ -25,14 +31,16 @@ export const usePreviewSmsTemplate = (locale?: string) => {
   });
 
   useEffect(() => {
-    getSmsPreview({
-      content: previewData.current.templateContent,
-      payload: '',
-      locale,
-    });
-  }, [locale, previewData, getSmsPreview]);
+    if (!showPreviewAsLoading) {
+      getSmsPreview({
+        content: previewData.current.templateContent,
+        payload: previewData.current.processedVariables,
+        locale,
+      });
+    }
+  }, [locale, previewData, getSmsPreview, showPreviewAsLoading]);
```

Comment:
trigger use effect when showPreviewAsLoading is toggled, and call the preview endpoint only when typing in editor has timed out